### PR TITLE
Improve configurable BBE

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

## Goals
Improve the configurable BBE with latest changes

## Approach
N/A

## User stories
N/A

## Release note
N/A

## Certification
N/A

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
jdk 11.0.9 , ubuntu 20.04

### CODE REVIEW

File: examples/configurable/configurable.out [hunk 1]
```diff
@@ -32,3 +31,15 @@ maximum payload (in MB): 0.321
 remote enabled: false
 admin details: {"username":"admin","password":"password"}
 users: [{"username":"John","password":"abc123"},{"username":"Bob","password":"cde456"}]
+
+# A similar configuration can be given via CLI parameters in the
+# following format.
+
+bal run configurable.bal -- -Chost=192\.168\.1\.1 -Cport=9091 -CenableRemote=false -CmaxPayload=0.456
```
Comment (manuranga) [inline examples/configurable/configurable.out:38]: @warunalakshitha do we need to escape the dots?

Comment (HindujaB) [reply examples/configurable/configurable.out]: It seems we don't need to escape `.` character in CLI. Removed it in latest commit.

File: examples/configurable/configurable.description [hunk 2]
```diff
@@ -3,12 +3,32 @@
...
-// Ballerina currently supports configurable variables of types `int`, `float`,
-// `boolean`, `string`, `decimal`, the arrays of the respective types, record, and
-// table.
+// The values for `configurable` variables can be provided through configuration files,
+// CLI parameters, and environment variables.
+// The precedence order for retrieving configurable values is as follows:
+//
+// * CLI parameters with the following format
+// ```
+// -Ckey=value
```
Comment (warunalakshitha) [inline examples/configurable/configurable.description:12]: Shall we add the key format as well. 

key:= [[orgname .] moduleName .] variable
orgname and module name is optional if the variable defined in the root module.

File: examples/configurable/configurable.description [hunk 3]
```diff
@@ -3,12 +3,32 @@
...
+// `int`, `float`, `boolean`, `string`, `decimal`, and `xml`.
+//
+// * Configuration files
+//
+// The values can be provided through configuration files in TOML(v0.4) format.
+// The file location can be set through an environment variable with
+// the name `BAL_CONFIG_FILES`. It is possible to specify multiple configuration files
+// using the OS-specific separator.
+// The file precedence order will be as specified in the environment variable.
+// If an environment variable is not specified, it is located in the current directory
```
Comment (warunalakshitha) [inline examples/configurable/configurable.description:24]: Do we need to support of providing toml content through string?

Comment (HindujaB) [reply examples/configurable/configurable.description:32]: I added it under the ` Environment variables` section. Should I move it under this section?

File: examples/configurable/configurable.bal [hunk 4]
```diff
@@ -8,34 +8,35 @@ type UserInfo record {|
...
+// A value must be supplied for such variables.
 configurable string hostName = ?;
 configurable int port = ?;
 
-// A `configurable` variable can be initialized with an expression that is not `?`.
-// The `Config.toml` file does not have to specify a value for such a variable.
-// But if specified, the value in the `Config.toml` file overrides the value
+// A `configurable` variable can be initialized with an expression that is
+// not `?`. It is not mandatory to specify a value for such a variable.
+// But if specified, the value provided at configuration overrides the value
```
Comment (praneesha) [inline examples/configurable/configurable.bal:17]:
```suggestion
// However, if specified, the value provided at configuration overrides the value
```

File: examples/configurable/configurable.description [hunk 5]
```diff
@@ -3,12 +3,40 @@
...
+// -Ckey=value
+// ```
+// The key of a CLI parameter can be specified as,
+//
+// ```
+// key:= [[org-name .] module-name .] variable
+// ```
+//
+// The org-name and module-name is optional for the variable defined in the
+// root module or single ballerina file. Currently, CLI based configuration
```
Comment (praneesha) [inline examples/configurable/configurable.description:21]:
```suggestion
// root module or single Ballerina file. Currently, CLI-based configuration
```

File: examples/configurable/configurable.description [hunk 6]
```diff
@@ -3,12 +3,40 @@
...
+// is supported for configurable variables of types `int`, `float`, `boolean`,
+// `string`, `decimal`, and `xml`.
+//
+// * Configuration files
+//
+// The values can be provided through configuration files in TOML(v0.4) format.
```
Comment (praneesha) [inline examples/configurable/configurable.description:27]:
```suggestion
// The values can be provided through configuration files in the TOML(v0.4) format.
```

File: examples/configurable/configurable.description [hunk 7]
```diff
@@ -3,12 +3,40 @@
...
+// the name `BAL_CONFIG_FILES`. It is possible to specify multiple configuration files
+// using the OS-specific separator.
+// The file precedence order will be as specified in the environment variable.
+// If an environment variable is not specified, it is located in the current directory
+// with the file name `Config.toml`.
+//
+// * Environment variables
+//
+// The values can be provided through an environment variable with the name
+// `BAL_CONFIG_DATA` where the content is expected to be in TOML(v0.4) format.
```
Comment (praneesha) [inline examples/configurable/configurable.description:38]:
```suggestion
// `BAL_CONFIG_DATA` in which the content is expected to be in the TOML(v0.4) format.
```

File: examples/configurable/configurable.description [hunk 8]
```diff
@@ -3,12 +3,40 @@
...
+//
+// Currently, TOML based configuration is supported for configurable variables of
```
Comment (praneesha) [inline examples/configurable/configurable.description:40]:
```suggestion
// Currently, TOML-based configuration is supported for configurable variables of
```

**Review by manuranga — APPROVED** [review]

**Pull request merged**